src/sage/schemes/elliptic_curves/ell_number_field.py

```diff
@@ -4067,3 +4073,183 @@ def rational_points(self, **kwds):
+            sage: E.is_modular(verbose=True)
+            All elliptic curves over QQ are modular
+            True
+            sage: K.<a> = QuadraticField(5)
+            sage: E = EllipticCurve([0,1,0,a,0])
+            sage: E.is_modular(verbose=True)
+            All elliptic curves over real quadratic fields are modular
+            True
+            sage: E = EllipticCurve([0,0,0,a,0])
+            sage: E.is_modular(verbose=True)
+            All elliptic curves over real quadratic fields are modular
+            True
+            sage: K.<a> = CyclotomicField(5)
+            sage: E = EllipticCurve([0,1,0,a,0])
+            sage: E.is_modular(verbose=True)
+            Unable to determine modularity except over totally real and imaginary quadratic fields
+            False
+
+        Some examples from the LMFDB.  Over a totally real quintic field::
+
+            sage: R.<x> = PolynomialRing(QQ)
+            sage: K.<a> = NumberField(R([1, 3, -1, -5, 0, 1]))
+            sage: E = EllipticCurve([K([3,0,-5,0,1]),K([-3,-7,9,2,-2]),K([1,0,0,0,0]),K([-22,-79,-92,14,21]),K([12,-87,-324,7,74])])
+            sage: E.is_modular(verbose=True)
+            Modular since the mod 3 Galois image is neither reducible nor split
+            True
+
+        Over imaginary quadratic fields `K` over which `X_0(15)` has
+        positive rank, there are many curves where current results do
+        not imply modularity::
+
+            sage: R.<x> = PolynomialRing(QQ)
+            sage: K.<a> = NumberField(R([2, -1, 1]))
+            sage: EllipticCurve('15a1').change_ring(K).rank()
+            1
+            sage: E = EllipticCurve([K([1,0]),K([1,-1]),K([1,1]),K([-105,98]),K([-615,1188])])
+            sage: E.is_modular(verbose=True)
+            Modularity not established: this curve has small image at 3 and 5
+            False
+            sage: EllipticCurve('15a1').change_ring(K).rank()
+            1
+
+        Nevertheless, over the same field, most curves pass::
+
+            sage: E = EllipticCurve([K([0,0]),K([0,1]),K([1,1]),K([-8,-3]),K([-5,-5])])
+            sage: E.is_modular(verbose=True)
+            Modular since the mod 3 Galois image is neither reducible nor split
+            True
+
+        Over a field which is neither totally real nor imaginary
+        quadratic, no conclusion is currently possible::
+
+            sage: R.<x> = PolynomialRing(QQ)
+            sage: K.<a> = NumberField(R([1, 0, -1, 1]))
+            sage: K.is_totally_real()
+            False
+            sage: E = EllipticCurve([K([0,0,1]),K([1,0,0]),K([0,0,1]),K([0,0,0]),K([0,0,0])])
+            sage: E.is_modular(verbose=True)
+            Unable to determine modularity except over totally real and imaginary quadratic fields
+            False
+        """
+        K = self.base_field()
+        d = K.degree()
+
+        if d == 1: # base field QQ
+            if verbose:
+                print("All elliptic curves over QQ are modular")
```

Comment:
See previous reply.  It would not be possible to give the citations in the verbose message without duplicating the citations I already put in, which would be very tedious to do.  Perhaps it is possible to give a URL for the official docstring in the reference manual on the sagemath website.